catalyst/dl/runner/core.py

```diff
@@ -364,23 +378,21 @@ def trace(
             predict_params=predict_params,
         )
 
-        if logdir is not None:
-            filename = utils.get_trace_name(
-                method_name=method_name,
-                mode=mode,
-                requires_grad=requires_grad,
-                opt_level=opt_level,
-            )
-
-            logdir = Path(logdir)
-            output: Path = logdir / "trace"
-            output.mkdir(exist_ok=True, parents=True)
+        # Restore previous state of the model
+        getattr(model, "train" if _is_training else "eval")()
+        utils.set_requires_grad(model, _requires_grad)
+        model.to(_device)
 
-            out_model = str(output / filename)
-
-            torch.jit.save(result, out_model)
+        utils.save_traced_model(
```

Comment:
should we check, that logdir is not none here?

Reply:
added several checks - one in `runner.trace` function, another in `utils.save_traced_model`